velox/common/memory/MemoryUsageTracker.h

```diff
@@ -120,116 +82,78 @@ class MemoryUsageTracker
+  /// reservation should be released with release(). If the new reserved amount
+  /// exceeds the usage limit, an exception will be thrown.
   void reserve(int64_t size);
 
-  // If a minimum reservation has been set with reserve(), resets the
-  // minimum reservation. If the current usage is below the minimum
-  // reservation, decreases reservation and usage down to the rounded
-  // actual usage.
-  void release();
-
-  // Increments outstanding memory by 'size', which is positive for
-  // allocation and negative for free. If there is no reservation or
-  // the new allocated amount exceeds the reservation, propagates the
-  // change upward.
-  void update(int64_t size);
+  /// Checks if it is likely that the reservation on 'this' can be incremented
+  /// by 'increment'. Returns false if this seems unlikely. Otherwise attempts
+  /// the reservation increment and returns true if succeeded.
+  bool maybeReserve(int64_t increment);
 
-  int64_t getCurrentUserBytes() const {
-    return adjustByReservation(user(currentUsageInBytes_));
-  }
-  int64_t getCurrentSystemBytes() const {
-    return adjustByReservation(system(currentUsageInBytes_));
-  }
-  int64_t getCurrentTotalBytes() const {
-    return adjustByReservation(
-        user(currentUsageInBytes_) + system(currentUsageInBytes_));
-  }
+  /// If a minimum reservation has been set with reserve(), resets the minimum
+  /// reservation. If the current usage is below the minimum reservation,
+  /// decreases reservation and usage down to the rounded actual usage.
+  void release();
 
-  int64_t getPeakUserBytes() const {
-    return user(peakUsageInBytes_);
-  }
-  int64_t getPeakSystemBytes() const {
-    return system(peakUsageInBytes_);
-  }
-  int64_t getPeakTotalBytes() const {
-    return total(peakUsageInBytes_);
-  }
+  /// Increments outstanding memory by 'size', which is positive for allocation
+  /// and negative for free. If there is no reservation or the new allocated
+  /// amount exceeds the reservation, propagates the change upward. Sometimes
+  /// the memory pool wants to mock an update for quota accounting purposes and
+  /// different memory usage trackers can choose to accommodate this
+  /// differently.
+  virtual void update(int64_t size);
 
-  int64_t getAvailableReservation() const {
-    return std::max<int64_t>(0, reservation_ - usedReservation_);
+  /// Returns the current memory usage.
+  int64_t currentBytes() const {
+    return adjustByReservation(currentBytes_);
   }
 
-  int64_t getNumAllocs() const {
-    return total(numAllocs_);
+  /// Returns the unused reservations in bytes.
+  int64_t availableReservation() const {
+    return std::max<int64_t>(0, reservationBytes_ - usedReservationBytes_);
   }
 
-  int64_t getCumulativeBytes() const {
-    return user(cumulativeBytes_) + system(cumulativeBytes_);
+  /// Returns the number of allocations.
+  int64_t numAllocs() const {
```

Comment:
Thank you for removing 'get' prefix from this and other getters. Would you also fix this name to avoid abbreviating: numAllocations().